src/test/java/com/epam/aidial/cfg/domain/validator/AddonValidatorTest.java

```diff
@@ -75,4 +78,22 @@ void validateAddonCreation_shouldThrowExceptionForInvalidName(String name) {
                 .hasMessageContaining("does not match the required pattern");
     }
 
+    @Test
+    void validateAddonCreation_shouldThrowExceptionWhenDeploymentValidatorThrows() {
+        // given
+        String deploymentName = "deploymentName";
+
+        Deployment deployment = new Deployment(deploymentName);
+
+        Addon addon = new Addon();
+        addon.setDeployment(deployment);
+
+        doThrow(IllegalArgumentException.class).when(deploymentValidator)
+                .validateCreation("Addon", deploymentName);
+
+        // when/then
+        Assertions.assertThatThrownBy(() -> addonValidator.validateAddonCreation(addon))
+                .isInstanceOf(IllegalArgumentException.class);
```

Comment:
i guess reason is empty name but not sure. reason is not obvious (add comment/rename test method/add assertion)./